include/libtorrent/kademlia/dht_tracker.hpp

```diff
@@ -101,11 +101,18 @@ namespace libtorrent { namespace dht
 			, boost::function<void(item const&, bool)> cb
 			, std::string salt = std::string());
 
+		// for immutable_item.
+		// the callback function will be called when put operation is done.
+		// the bool parameter indicates the put operation success or failed.
 		void put_item(entry data
-			, boost::function<void()> cb);
+			, boost::function<void(int)> cb);
 
+		// for mutable_item.
+		// the data_cb will be called when we get authoritative mutable_item,
+		// the cb is same as put immutable_item.
```

Comment:
I didn't see the try again logic. The original code even can't report error to user because the dht_put_alert is sent before put.